torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -733,7 +733,9 @@ def view_avoid_dupes_with_primals(t):
             traced_tangents=traced_tangents,
             subclass_inp_meta=create_subclass_meta(flat_args),
             subclass_fw_graph_out_meta=create_subclass_meta(fw_graph_outs),
-            subclass_tangent_meta=create_subclass_meta(traced_tangents),
+            subclass_tangent_meta=create_subclass_meta(
+                traced_tangents, count_symints=False
```

Comment:
Ok, I think I buy why you are explicitly setting `count_symints=False` here - the `tangent` arguments to the joint and backward fns do not take in any extra SymInts, so we don't want to include them in the corresponding metadata.

Since it's a bit complicated to get right - can we have some more assertions? One assertion in particular that I think would be nice is:

(1) add an extra bool onto `SubclassCreationMeta` that tags whether or not symints were included in its total `arg_count`, e.g. `included_subclass_symints=True/False`
(2) For every call-site to `wrap_tensor_subclasses()`, pass in an `include_subclass_symints=True/False` flag, where it is true in the cases where we expect extra symint arguments in the argument list
(3) add an assertion that `include_subclass_symints == included_subclass_symints`

Reply:
For item (3), should the assertion be `include_subclass_symints == meta.included_subclass_symints?`

Reply:
yep